maven-artifact/src/main/java/org/apache/maven/artifact/Artifact.java

```diff
@@ -86,6 +87,16 @@
 
     void setFile( File destination );
 
+    default Path getPath()
```

Comment:
You can test these two new default methods with a subclass that implements getFile and setFile but not getPath and setPath. You don't need mocks, and you certainly don't need spies to do that. Focus on the values returned by getPath, not which methods getPath and setPath call.

Reply:
Yes I *can* do that but I do not see any need to do that. My test tests exctly what I want to test, so I do not see *why* I should change it. I *do not want* to test what would happen *if* someone implements that, I *only* want to proof what the default methods *actually* are like.

Reply:
The reason is that it is *necessary* to test that `setPath` actually invokes `setFile` as this is *what I have coded* in the default method and what get executed one you do *not* implement `setPath`. And only THAT case is what I MUST test, everything is something *differen* that you can add in *another* test ontop.